File: datasets/lua/HumanEval_104_unique_digits.lua
-- Given a list of positive integers x. return a sorted list of all 
-- elements that hasn't any even digit.
-- Note: Returned list should be sorted in increasing order.
-- For example:
-- >>> unique_digits([15, 33, 1422, 1])
-- [1, 15, 33]
-- >>> unique_digits([152, 323, 1422, 10])
-- []
local function unique_digits(x)
  local result = {}
  for _, v in ipairs(x) do
    if is_unique(v) then
      table.insert(result, v)
    end
  end
  table.sort(result)
  return result
end
-- Unit tests below

lu = require('luaunit')

function test_humaneval()
local candidate = unique_digits
    lu.assertEquals(candidate({15, 33, 1422, 1}), {1, 15, 33})
    lu.assertEquals(candidate({152, 323, 1422, 10}), {})
    lu.assertEquals(candidate({12345, 2033, 111, 151}), {111, 151})
    lu.assertEquals(candidate({135, 103, 31}), {31, 135})
end

os.exit(lu.LuaUnit.run())

File: datasets/php/HumanEval_104_unique_digits.php
<?php
// Given a list of positive integers x. return a sorted list of all 
// elements that hasn't any even digit.
// Note: Returned list should be sorted in increasing order.
// For example:
// >>> unique_digits([15, 33, 1422, 1])
// [1, 15, 33]
// >>> unique_digits([152, 323, 1422, 10])
// []
function unique_digits($x) {
    $result = [];
    foreach ($x as $value) {
        if (strpos($value, '2') === false && strpos($value, '4') === false && strpos($value, '6') === false && strpos($value, '8') === false) {
            array_push($result, $value);
        }
    }
    return $result;
}


function candidate(...$args) {
    return unique_digits(...$args);
}

function test(): void {
    if (candidate(array(15, 33, 1422, 1)) !== array(1, 15, 33)) { throw new Exception("Test failed!"); }
    if (candidate(array(152, 323, 1422, 10)) !== array()) { throw new Exception("Test failed!"); }
    if (candidate(array(12345, 2033, 111, 151)) !== array(111, 151)) { throw new Exception("Test failed!"); }
    if (candidate(array(135, 103, 31)) !== array(31, 135)) { throw new Exception("Test failed!"); }
}

test();